src/charm.py

```diff
@@ -245,22 +282,31 @@ def _get_s3_env(self) -> typing.Dict[str, typing.Any]:
 
         return s3_env
 
-    def _create_discourse_environment_settings(self) -> typing.Dict[str, typing.Any]:
-        """Create a layer config based on our current configuration.
+    def _get_redis_relation_data(self) -> typing.Tuple[typing.Any, typing.Any]:
+        """Get the hostname and port from the redis relation data.
 
         Returns:
-            Dictionary with all the environment settings.
+            Tuple with the hostname and port of the related redis
         """
-        # Get redis connection information from the relation.
-        redis_hostname = None
-        redis_port = 6379
         # This is the current recommended way of accessing the relation data.
         for redis_unit in self._stored.redis_relation:  # type: ignore
-            redis_hostname = self._stored.redis_relation[redis_unit].get("hostname")  # type: ignore
-            redis_port = self._stored.redis_relation[redis_unit].get("port")  # type: ignore
+            # mypy fails to see that this is indexable
+            redis_unit_data = self._stored.redis_relation[redis_unit]  # type: ignore
+            redis_hostname = redis_unit_data.get("hostname", "")  # type: ignore
+            redis_port = redis_unit_data.get("port", DEFAULT_REDIS_PORT)  # type: ignore
             logger.debug(
                 "Got redis connection details from relation of %s:%s", redis_hostname, redis_port
             )
+        return (redis_hostname, redis_port)
+
+    def _create_discourse_environment_settings(self) -> typing.Dict[str, typing.Any]:
```

Comment:
Nitpick:
I'm wondering if the `database_relation_data` and `redis_relation_data` would be better instantiated on the handler and be passed to environment settings as variables. I think it could help understand the dataflow better.
I understand that this could be a stylistic preference though.

Reply:
I want to leave what is related to redis in the charm (for the scope of this PR at least), since the database module will be about postgres only.

It's `get_relation_data()` that creates `database_relation_data`, so it's already in the database module. I'm not sure to follow what you mean

Reply:
I was thinking instead of 
```
def _create_discourse_environment_settings(self)
    database_relation_data = self._database.get_relation_data()
    redis_relation_data = self._get_redis_relation_data()
```,
it would be better to do 
```
def _create_discourse_environment_settings(self, db_relation_data, redis_relation_data)
    ...
```
so that the dataflow is singular rather than being referred / called in different places within the execution flow. However, it's a minor stylistic difference. Would this explanation be a bit clearer?
Btw feel free to resolve this, I don't think it's important change within this scope! :D

Reply:
I see.  
But here, it's the only place in the whole charm where I refer to get_relation_data(), so it's already singular (and it would not be with your proposal, I would have to add it at 5 different places).

So I'm going to leave it as is